parquet-column/src/test/java/org/apache/parquet/schema/TestPrimitiveComparator.java

```diff
@@ -0,0 +1,286 @@
+  }
+
+  @Test
+  public void testUnsignedInt64Comparator() {
+    testInt64Comparator(UNSIGNED_INT64_COMPARATOR,
+      0,
+      1,
+      12345678901L,
+      Long.MAX_VALUE,
+      Long.MIN_VALUE,
+      -12345678901L,
+      -1);
+  }
+
+  private void testInt64Comparator(PrimitiveComparator<Long> comparator, long... values) {
+    for (int i = 0; i < values.length; ++i) {
+      for (int j = 0; j < values.length; ++j) {
+        long vi = values[i];
+        long vj = values[j];
+        int exp = i - j;
+        assertSignumEquals(vi, vj, exp, comparator.compare(vi, vj));
+        assertSignumEquals(vi, vj, exp, comparator.compare(Long.valueOf(vi), Long.valueOf(vj)));
+      }
+    }
+
+    checkThrowingUnsupportedException(comparator, Long.TYPE);
+  }
+
+  @Test
+  public void testFloatComparator() {
+    float[] values = {
+      Float.NEGATIVE_INFINITY,
+      -Float.MAX_VALUE,
+      -1234.5678F,
+      -Float.MIN_VALUE,
+      0,
+      Float.MIN_VALUE,
+      1234.5678F,
+      Float.MAX_VALUE,
+      Float.POSITIVE_INFINITY};
+
+    for (int i = 0; i < values.length; ++i) {
+      for (int j = 0; j < values.length; ++j) {
+        float vi = values[i];
+        float vj = values[j];
+        int exp = i - j;
+        assertSignumEquals(vi, vj, exp, FLOAT_COMPARATOR.compare(vi, vj));
+        assertSignumEquals(vi, vj, exp, FLOAT_COMPARATOR.compare(Float.valueOf(vi), Float.valueOf(vj)));
+      }
+    }
+
+    checkThrowingUnsupportedException(FLOAT_COMPARATOR, Float.TYPE);
+  }
+
+  @Test
+  public void testDoubleComparator() {
+    double[] values = {
+      Double.NEGATIVE_INFINITY,
+      -Double.MAX_VALUE,
+      -123456.7890123456789,
+      -Double.MIN_VALUE,
+      0,
+      Double.MIN_VALUE,
+      123456.7890123456789,
+      Double.MAX_VALUE,
+      Double.POSITIVE_INFINITY};
+
+    for (int i = 0; i < values.length; ++i) {
+      for (int j = 0; j < values.length; ++j) {
+        double vi = values[i];
+        double vj = values[j];
+        int exp = i - j;
+        assertSignumEquals(vi, vj, exp, DOUBLE_COMPARATOR.compare(vi, vj));
+        assertSignumEquals(vi, vj, exp, DOUBLE_COMPARATOR.compare(Double.valueOf(vi), Double.valueOf(vj)));
+      }
+    }
+
+    checkThrowingUnsupportedException(DOUBLE_COMPARATOR, Double.TYPE);
+  }
+
+  @Test
+  public void testLexicographicalBinaryComparator() {
+    testObjectComparator(LEXICOGRAPHICAL_BINARY_COMPARATOR,
+      Binary.fromConstantByteArray(new byte[0]),
+      Binary.fromConstantByteArray(new byte[]{127, 127, 0, 127}, 2, 1),
+      Binary.fromCharSequence("aaa"),
+      Binary.fromString("aaaa"),
+      Binary.fromReusedByteArray("aaab".getBytes()),
+      Binary.fromReusedByteArray("azzza".getBytes(), 1, 3),
+      Binary.fromReusedByteBuffer(ByteBuffer.wrap("zzzzzz".getBytes())),
+      Binary.fromReusedByteBuffer(ByteBuffer.wrap("aazzzzzzaa".getBytes(), 2, 7)),
+      Binary.fromConstantByteBuffer(ByteBuffer.wrap(new byte[]{-128, -128, -128})),
```

Comment:
Hexadecimal literals in Java are integers which means while `-128` does not required casting to byte `0x80` would. Do you think it would still be readable if I write `new byte[]{(byte) 0x80, (byte) 0x80, (byte) 0xFF}`?

Reply:
You're right. Optional: Please consider adding the hex values in comments, like // 0x8080FF

Reply:
I prefer the option with casting for small values like these.